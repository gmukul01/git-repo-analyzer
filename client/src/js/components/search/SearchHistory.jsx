import React, { PureComponent } from "react";
import Modal from "components/Modal";

export default class SearchHistory extends PureComponent {
  render() {
    const { history } = this.props;

    const rows = history.map((repo, index) => {
      return (
        <tr key={repo.updated_at}>
          <td>{repo.repoName}</td>
          <td>{repo.commitCount}</td>
          <td>{repo.openPullRequenCount}</td>
          <td>
            <a href={`${repo.htmlUrl}`} target="_blank">
              Link
            </a>
          </td>
          <td>
            <a href="#" data-toggle="modal" data-target={`#${index}`}>
              Readme
            </a>
          </td>
        </tr>
      );
    });

    const modals = history.map((repo, index) => {
      return <Modal key={repo.updated_at} readme={repo.readme} modalId={`${index}`} modalHeader={repo.repoName} />;
    });

    return (
      <section className="table-responsive search-history-table">
        <table className="table table-hover">
          <thead>
            <tr>
              <th>Repo Full Name</th>
              <th>Commits</th>
              <th>Open Pull Reuests</th>
              <th>Repo Link</th>
              <th>Readme</th>
            </tr>
          </thead>
          <tbody>{rows}</tbody>
        </table>
        {modals}
      </section>
    );
  }
}
