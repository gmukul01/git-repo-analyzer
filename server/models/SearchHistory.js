import mongoose from "mongoose";
const Schema = mongoose.Schema;

const SearchHistorySchema = new Schema({
  user: {
    type: Schema.Types.ObjectId,
    ref: "users"
  },
  history: [
    {
      repoName: String,
      htmlUrl: String,
      updated_at: {
        type: Date,
        default: Date.now
      }
    }
  ]
});

export default mongoose.model("search-histories", SearchHistorySchema);
