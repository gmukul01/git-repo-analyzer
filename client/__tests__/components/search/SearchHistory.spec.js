import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";

import SearchHistory from "components/search/SearchHistory";

describe("SearchHistory Component", () => {
  const initialProps = {
    history: [
      {
        repoName: "dummyRepoName1",
        commitCount: 10,
        openPullRequenCount: 1,
        htmlUrl: "dummyUrl1",
        updated_at: "updated_at1"
      },
      {
        repoName: "dummyRepoName2",
        commitCount: 10,
        openPullRequenCount: 1,
        htmlUrl: "dummyUrl2",
        updated_at: "updated_at2"
      }
    ]
  };

  it("should match snapshot", () => {
    const renderer = new ShallowRenderer(),
      result = renderer.render(<SearchHistory {...initialProps} />);
    expect(result).toMatchSnapshot();
  });
});
