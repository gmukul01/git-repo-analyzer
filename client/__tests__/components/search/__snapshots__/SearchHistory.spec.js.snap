// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHistory Component should match snapshot 1`] = `
<section
  className="table-responsive search-history-table"
>
  <table
    className="table table-hover"
  >
    <thead>
      <tr>
        <th>
          Repo Full Name
        </th>
        <th>
          Commits
        </th>
        <th>
          Open Pull Reuests
        </th>
        <th>
          Repo Link
        </th>
        <th>
          Readme
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          dummyRepoName1
        </td>
        <td>
          10
        </td>
        <td>
          1
        </td>
        <td>
          <a
            href="dummyUrl1"
            target="_blank"
          >
            Link
          </a>
        </td>
        <td>
          <a
            data-target="#0"
            data-toggle="modal"
            href="#"
          >
            Readme
          </a>
        </td>
      </tr>
      <tr>
        <td>
          dummyRepoName2
        </td>
        <td>
          10
        </td>
        <td>
          1
        </td>
        <td>
          <a
            href="dummyUrl2"
            target="_blank"
          >
            Link
          </a>
        </td>
        <td>
          <a
            data-target="#1"
            data-toggle="modal"
            href="#"
          >
            Readme
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <Modal
    modalHeader="dummyRepoName1"
    modalId="0"
  />
  <Modal
    modalHeader="dummyRepoName2"
    modalId="1"
  />
</section>
`;
